src/box/memtx_tx.c

```diff
@@ -1791,9 +1791,8 @@ memtx_tx_handle_gap_write(struct txn *txn, struct space *space,
 	struct rlist *fsc_list = &index->full_scans;
 	uint64_t index_mask = 1ull << (ind & 63);
 	rlist_foreach_entry_safe(fsc_item, fsc_list, in_full_scans, fsc_tmp) {
-		if (fsc_item->txn != txn)
-			memtx_tx_track_read_story(fsc_item->txn, space, story,
-						  index_mask);
+		memtx_tx_track_read_story(fsc_item->txn, space, story,
```

Comment:
No, it's not reverting of 3e546d4. I added to the commit message:
```
Another similar mistake was made in e6f5090ce7, where writing
after full scan of the same transaction was not tracked as read.
Obviously that was wrong: if some other transaction overwrites
the key and commits - this transaction must go to read view since
it did not see anything by this key which is not so anymore.
```